### PR TITLE
Bugfix/ker/minor fixes

### PR DESCRIPTION
**Pull request by keriksson-rosenqvist**
Collated minor fixes ported from main:
- Test raise TypeError when executing unsupported type.
- Enable skipped complex gate test.
- Ensure sweep iterators are always reverted.
- Extend test_load_hardware_definition to different qubit_counts, and temp fix reursion issue.
- Convert test_calibration.py to pytest format.
- Fix minor typo.
- Improved QuantumInstruction add_channel functionality.
- Removed duplicate numpy import.
- Allowing the hardware to work without a control hardware.
- Adding hardcoded delay to qubit drives and counting two clock cycles per instruction for the sequencer.
- Wh/switch hive fix.
- Fix frequency frame attribute access.

### CODE REVIEW

File: src/QAT/qat/purr/backends/live_devices.py [hunk 1]
```diff
@@ -74,10 +74,11 @@ class ControlHardwareChannel(PhysicalChannel):
     derived) object should contain hardware specific information.
     """
 
-    def __init__(self, id_, hardware_id, dcbiaschannel_pair, *args, **kwargs):
+    def __init__(self, id_, hardware_id, dcbiaschannel_pair, *args, switch_ch=None, **kwargs):
```
Comment (jfriel-oqc) [inline src/QAT/qat/purr/backends/live_devices.py:77]: this should be before *args

**Review by jfriel-oqc — APPROVED** [review]
approving up to that one argument order swtich

File: src/QAT/qat/purr/backends/live_devices.py (hunk 1 above)
Comment (keriksson-rosenqvist) [reply src/QAT/qat/purr/backends/live_devices.py:77]: It's a `kwarg`, why would it be before `*args`?

Comment (jfriel-oqc) [reply src/QAT/qat/purr/backends/live_devices.py:77]: ah you're right, sorry! Misread with test wrapping

**Pull request merged**